### PR TITLE
[IMP] core: reporting of browser logs / errors during setup

### PR DESCRIPTION
**Pull request by xmo-odoo**
Improve handling of JS logging in headless runner in order to avoid losing logs
and errors e.g. the issue fixed by odoo/odoo#41231 passed because it occurred
during module loading, which happens during initial page loading (browser_js >
navigate_to > _websocket_wait_event), which ignored logs (and exceptions though
here it's a console.error log), and as a result reported no failure (and would
simply miss that specific test as well as every test following it).

This requires additional modifications as we have a fair amount of silent
failures (exceptions or logging.error calls) at the moment:

- downgrade one error to a warning (which becomes an info at the python level)
- cleanup some synthetic / mock errors to better match what comes over RPC (and avoid transient or setup failures)
- tours which end in an action, leading to JS code executing during browser cleanup (and blowing up)
  - if the last step triggers a default (implicit) run, replace by a no-op
  - fix tours for which that does not work by either modifying the last step or adding an additional check step

### CODE REVIEW

File: addons/bus/static/src/js/crosstab_bus.js [hunk 1]
```diff
@@ -342,7 +342,7 @@ var CrossTabBus = Longpolling.extend({
      */
     _onUnload: function () {
         // unload peer
-        var peers = this._callLocalStorage('getItem', 'peers', {});
+        var peers = this._callLocalStorage('getItem', 'peers') || {};
         delete peers[this._id];
```
Comment (xmo-odoo) [inline addons/bus/static/src/js/crosstab_bus.js]: @VincentSchippefilt after improving logging handling so we get proper error reporting of loading issues on runbot, it reports an issue here with "cannot convert undefined or null to object", I tried finding out why but got ~~high~~ lost amongst the indirection.

Even with the logging improvements of this PR, the debugging experience is not exactly stellar, and this issue doesn't seem to occur when running in Chrome/Chromium (and obviously nothing works in firefox). So this fixes the symptoms but not the underlying issue that there are apparently situations where the default-setting simply doesn't work. I'd guess an issue with the callback being called asynchronously in https://github.com/odoo/odoo/blob/236b8f120bb23aa569397cf4dc6b881d6f41155a/addons/web/static/src/js/core/service_mixins.js#L113-L125

Comment (VincentSchippefilt) [reply addons/bus/static/src/js/crosstab_bus.js]: @xmo-odoo @alexkuhn I'm forwarding to AKU, for crosstab bussing things ;-)

Comment (alexkuhn) [reply addons/bus/static/src/js/crosstab_bus.js]: The correct fix has been applied in 13.1, could be backported in 13.0:
https://github.com/odoo/odoo/commit/597f12a7578f14b483260ddb941ff240feccfd35

Comment (alexkuhn) [reply addons/bus/static/src/js/crosstab_bus.js]: nevermind... it has been fixed in 12.0 and above :)

File: addons/bus/static/src/js/crosstab_bus.js [hunk 2]
```diff
@@ -348,7 +348,7 @@ var CrossTabBus = Longpolling.extend({
      */
     _onUnload: function () {
         // unload peer
-        var peers = this._callLocalStorage('getItem', 'peers') || {};
+        var peers = this._callLocalStorage('getItem', 'peers', {});
```
Comment (xmo-odoo) [inline addons/bus/static/src/js/crosstab_bus.js]:
```suggestion
        var peers = this._callLocalStorage('getItem', 'peers') || {};
```

**Comment by xmo-odoo** [comment]
@VincentSchippefilt did some more updates, branch finally seems to pass. Even after rebasing on 13.0 reverting most JS changes (probably aside from the awaits, I don't know if I checked all of them) makes the build fail.

File: addons/web/tests/test_js.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ class WebSuite(odoo.tests.HttpCase):
 
     def test_js(self):
         # webclient desktop test suite
-        self.phantom_js('/web/tests?mod=web&failfast', "", "", login='admin', timeout=1800)
+        self.phantom_js('/web/tests?mod=web&failfast&debug=assets', "", "", login='admin', timeout=1800)
```
Comment (VincentSchippefilt) [inline addons/web/tests/test_js.py]: Do you really want to start the tests in debug=assets, there are some difference in behavior.

Comment (xmo-odoo) [reply addons/web/tests/test_js.py]: Not necessarily, it's just that debugging tests through headless chrome without this is hell.

**Comment by xmo-odoo** [comment]
ping @VincentSchippefilt

**Comment by VincentSchippefilt** [comment]
in PR message: 

> Targets 13.0 rather than 13.0 because the code was sufficiently modified between 12 and 13 (without backporting) that it's a pain to apply the change.

You ment 13.0 rather than 12.0 ?

**Comment by xmo-odoo** [comment]
> in PR message:
> 
> > Targets 13.0 rather than 13.0 because the code was sufficiently modified between 12 and 13 (without backporting) that it's a pain to apply the change.
> 
> You ment 13.0 rather than 12.0 ?

Either that or alternate-reality 13.0

**Comment by VincentSchippefilt** [comment]
I'm good for the JS changes (and for the alternate reality 13.0).  I'm not confident to review the python part.

**Comment by xmo-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by xmo-odoo** [comment]
@robodoo rebase-ff

**Comment by xmo-odoo** [comment]
@robodoo retry

**Comment by xmo-odoo** [comment]
@robodoo retry

**Pull request closed without merging**